examples/signals/src/lib.rs

```diff
@@ -0,0 +1,131 @@
+use gdnative::*;
+
+#[derive(NativeClass)]
+#[inherit(Node)]
+// register_with attribute can be used to specify custom register function for node signals and properties
+#[register_with(Self::register_signals)]
+struct SignalEmitter {
+    timer: f64,
+    data: i64,
+}
+
+#[methods]
+impl SignalEmitter {
+    fn register_signals(builder: &init::ClassBuilder<Self>) {
+        builder.add_signal(init::Signal {
+            name: "tick",
+            args: &[],
+        });
+
+        builder.add_signal(init::Signal {
+            name: "tick_with_data",
+            // this predefined list of signal arguments is optional.
+            // Editor needs it to generate correct argument lists from UI.
+            // Signal will work as good as previous even with empty args
```

Comment:
English wording suggestion (replaces the three lines):

```suggestion
            // Argument list used by the editor for GUI and generation of GDScript handlers. It can be omitted if the signal is only used from code.
```